### PR TITLE
Troubleshooting pub socket error

### PR DESCRIPTION
**Pull request by jonasfj**
Some lessons from https://github.com/dart-lang/pub-dev/issues/4690

### CODE REVIEW

File: src/tools/pub/troubleshoot.md [hunk 1]
```diff
@@ -123,3 +123,23 @@ Try this workaround, which uses the command line to complete sign-in:
...
+```
+
+This can happen for many reasons, check your internet connection, and verify that
+you do not have firewalls or other security software that block internet access from `dart`.
+
+Users have reported issues using `dart pub` along with:
+ * `Kaspersky Internet Security`
+
+It might be possible that `dart pub` can be used along side such firewalls,
+if firewalls are correctly configured.
```
Comment (jonasfj) [inline src/tools/pub/troubleshoot.md:145]: I'm not sure if there is a good way to warn users, but it seems that many of these security products contain multiple security features. And it's easy for a user to disable one of the security features (thinking they've disabled all), observe that disabling it didn't resolve the issue for them, and thus, conclude that it is not their security software that is interfering.

Comment (EtienneBauscher) [reply src/tools/pub/troubleshoot.md:145]: @jonasfj - I must apologise to you and István Soós, I installed the Security Software without configuring trusted zones. I reinstalled it now and added `https://pub.dartlang.org` to the trusted zone and `flutter pub get` ran without a problem. My humble apologies for the inconvenience caused, it was due to my own inexperience.

Comment (jonasfj) [reply src/tools/pub/troubleshoot.md:145]: @EtienneBauscher could you provide a few hints on how to configure a trust zone.

Then maybe we change this section to say:
> Configuring trust zones for ...

Or if you publish those steps as a gist we can just link there for people using this software..

It's better than some vague comments saying that some users had trouble using this with that..

Comment (EtienneBauscher) [reply src/tools/pub/troubleshoot.md:145]: @jonasfj - Will the below suffice?

https://gist.github.com/EtienneBauscher/5d8cefe797e0400dd001d461ecc6d904

Comment (kwalrath) [reply src/tools/pub/troubleshoot.md:145]: @jonasfj did you plan to incorporate some of Etienne's text?

Comment (kwalrath) [reply src/tools/pub/troubleshoot.md:145]: @jonasfj do you plan to update the trust zone info?

Comment (jonasfj) [reply src/tools/pub/troubleshoot.md]: Yeah, this seems useful, I've cleaned up a bit and included it as an expandable section.

If we get other reports on how to configure firewalls, we can add the instructions here.

**Comment by kwalrath** [comment]
I've staged my suggested (and committed :) changes:

https://dartlang-org-staging-0.firebaseapp.com/tools/pub/troubleshoot#pub-get-socket-error

**Review by kwalrath — APPROVED** [review]

**Review by parlough — APPROVED** [review]

**Comment by kwalrath** [comment]
@jonasfj are you OK with my changes?

**Pull request merged**